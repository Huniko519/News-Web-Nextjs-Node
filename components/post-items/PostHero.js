import React from 'react';
import PropTypes from 'prop-types';
import PostItemMedia from './PostItemMedia';
import PostItemBadge from './PostItemBadge';
import PostItemTitle from './PostItemTitle';
import PostItemLabel from './PostItemLabel';
import Qoute from '../../utils/Qoute';

/**
 *
 * ╔══════════════════╗
 * ║                  ║
 * ║      Image       ║
 * ║                  ║
 * ╟──────────────────╢
 * ║     Category     ║
 * ║     Headline     ║
 * ║     Excerpt      ║
 * ╚══════════════════╝
 *
 * @param {post: {...postData}} props
 */
const PostHero = (props) => {
  const { post } = props;
  const isSponsored = post.type === 'post_sponsored';

  return (
    <div
      className={`inews__post inews__post-hero ${post.template ? post.template : ''}`}
      key={post.id}
      data-post-id={post.id}
    >

      <PostItemMedia
        className="inews__post-hero__media"
        imageSize="medium-16:9"
        width={16}
        height={9}
        layout="responsive"
        placeholder="/static/images/placeholder/placeholder-640x360.png"
        post={post}
      />
      <div className="inews__post-hero__content">
        <PostItemBadge post={post} showAuthor={post.template === 'opinion'} />
        <div className="inews__post-hero__headline">
          {post.template === 'opinion' && <Qoute size="medium" />}
          <PostItemLabel post={post} />
          <PostItemTitle post={post} />
          {isSponsored && (
            <span className="inews__post__sponsored-label">
              Promoted Content
            </span>
          )}
          <p dangerouslySetInnerHTML={{ __html: post.excerpt.rendered }} />
        </div>
      </div>
    </div>
  );
};

/* istanbul ignore next */
PostHero.propTypes = {
  post: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default PostHero;
