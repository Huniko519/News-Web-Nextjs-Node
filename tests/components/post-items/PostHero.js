import React from 'react';
import PostHero from '../../../components/post-items/PostHero';

test('renders the snapshot', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postHero = shallow(<PostHero post={post} />);
  expect(postHero).toMatchSnapshot();
});

it('should output with the correct class on the root element', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postHero = shallow(<PostHero post={post} />);
  expect(postHero.hasClass('inews__post')).toBe(true);
  expect(postHero.hasClass('inews__post-hero')).toBe(true);
});

it('should call the <PostItemMedia /> component with the correct props', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };

  const postHero = shallow(<PostHero post={post} />);
  const postItemMedia = postHero.find('PostItemMedia');
  expect(postItemMedia.length).toBe(1);
  expect(postItemMedia.prop('className')).toBe('inews__post-hero__media');
  expect(postItemMedia.prop('imageSize')).toBe('medium-16:9');
  expect(postItemMedia.prop('width')).toBe(16);
  expect(postItemMedia.prop('height')).toBe(9);
  expect(postItemMedia.prop('layout')).toBe('responsive');
  expect(postItemMedia.prop('placeholder')).toBe('/static/images/placeholder/placeholder-640x360.png');
  expect(postItemMedia.prop('post')).toMatchObject(post);
});

it('should call the <PostItemTitle /> component with the correct props', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };

  const postHero = shallow(<PostHero post={post} />);
  const postItemTitle = postHero.find('PostItemTitle');
  expect(postItemTitle.length).toBe(1);
  expect(postItemTitle.prop('post')).toMatchObject(post);
});

it('should output a Sponsored label if post.type === post_sponsored', () => {
  const post = {
    id: 1,
    type: 'post_sponsored',
    title: {
      rendered: 'title',
    },
    excerpt: {
      rendered: 'excerpt',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postHero = shallow(<PostHero post={post} />);
  const label = postHero.find('span.inews__post__sponsored-label');
  expect(label.length).toBe(1);
  expect(label.text()).toBe('Promoted Content');
});
