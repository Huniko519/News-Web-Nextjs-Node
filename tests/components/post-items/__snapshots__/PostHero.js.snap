// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the snapshot 1`] = `
<div
  className="inews__post inews__post-hero "
  data-post-id={1}
  key="1"
>
  <PostItemMedia
    className="inews__post-hero__media"
    height={9}
    imageSize="medium-16:9"
    layout="responsive"
    linkToAuthor={false}
    placeholder="/static/images/placeholder/placeholder-640x360.png"
    post={
      Object {
        "breadcrumbs": Array [
          Object {
            "name": "News",
          },
        ],
        "excerpt": Object {
          "rendered": "excerpt",
        },
        "id": 1,
        "link": "https://inews-article.com/test-article",
        "title": Object {
          "rendered": "title",
        },
      }
    }
    postItemType=""
    showAuthor={false}
    width={16}
  />
  <div
    className="inews__post-hero__content"
  >
    <PostItemBadge
      post={
        Object {
          "breadcrumbs": Array [
            Object {
              "name": "News",
            },
          ],
          "excerpt": Object {
            "rendered": "excerpt",
          },
          "id": 1,
          "link": "https://inews-article.com/test-article",
          "title": Object {
            "rendered": "title",
          },
        }
      }
      showAuthor={false}
    />
    <div
      className="inews__post-hero__headline"
    >
      <PostItemLabel
        post={
          Object {
            "breadcrumbs": Array [
              Object {
                "name": "News",
              },
            ],
            "excerpt": Object {
              "rendered": "excerpt",
            },
            "id": 1,
            "link": "https://inews-article.com/test-article",
            "title": Object {
              "rendered": "title",
            },
          }
        }
      />
      <PostItemTitle
        post={
          Object {
            "breadcrumbs": Array [
              Object {
                "name": "News",
              },
            ],
            "excerpt": Object {
              "rendered": "excerpt",
            },
            "id": 1,
            "link": "https://inews-article.com/test-article",
            "title": Object {
              "rendered": "title",
            },
          }
        }
      />
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "excerpt",
          }
        }
      />
    </div>
  </div>
</div>
`;
